server/src/test/java/org/opensearch/gateway/remote/RemoteGlobalMetadataManagerTests.java

```diff
@@ -480,14 +481,21 @@ public void testGetAsyncWriteRunnable_TemplatesMetadata() throws Exception {
     }
 
     public void testGetAsyncReadRunnable_CustomMetadata() throws Exception {
+        for (Version version : List.of(Version.CURRENT, Version.V_2_15_0, Version.V_2_13_0)) {
+            verifyCustomMetadataReadForVersion(version);
+        }
+    }
+
+    private void verifyCustomMetadataReadForVersion(Version version) throws Exception {
         Metadata.Custom customMetadata = getCustomMetadata();
         String fileName = randomAlphaOfLength(10);
         RemoteCustomMetadata customMetadataForDownload = new RemoteCustomMetadata(
             fileName,
             IndexGraveyard.TYPE,
             CLUSTER_UUID,
             compressor,
-            namedWriteableRegistry
+            namedWriteableRegistry,
+            version
         );
         when(blobStoreTransferService.downloadBlob(anyIterable(), anyString())).thenReturn(
             customMetadataForDownload.customBlobStoreFormat.serialize(customMetadata, fileName, compressor).streamInput()
```

Comment:
in this case it will serialize with which version?

Reply:
serialize will always happen with current version only